instrumentation/afl-compiler-rt.o.c

```diff
@@ -629,6 +630,30 @@ static void __afl_unmap_shm(void) {
 
 }
 
+void write_error(char *text) {
+
+  u8 *  o = getenv("__AFL_OUT_DIR");
+  char *e = strerror(errno);
+
+  if (o) {
+
+    char buf[4096];
+    snprintf(buf, sizeof(buf), "%s/error.txt", o);
+    FILE *f = fopen(buf, "a");
+
+    if (f) {
+
+      fprintf(f, "Error(%s): %s\n", text, e);
+      fclose(f);
+
+    }
+
+  }
+
+  fprintf(stderr, "Error(%s): %s\n", text, e);
+
+}
+
```

Comment:
How about we use some macro like WARNF(), which also automatically tells the origin (file and line number). Then the analysis would be easier, I think.

Reply:
the WARNF location would be in the print_error() function, that is not helpful

Reply:
Think macro instead of function like this:
```
--- a/instrumentation/afl-compiler-rt.o.c
+++ b/instrumentation/afl-compiler-rt.o.c
@@ -630,7 +630,9 @@ static void __afl_unmap_shm(void) {
 
 }
 
-void write_error(char *text) {
+#define write_error(text) write_error_with_location(text, __FILE__, __LINE__)
+
+void write_error_with_location(char *text, char* filename, int linenumber) {
 
   u8 *  o = getenv("__AFL_OUT_DIR");
   char *e = strerror(errno);
@@ -643,14 +645,14 @@ void write_error(char *text) {
 
     if (f) {
 
-      fprintf(f, "Error(%s): %s\n", text, e);
+      fprintf(f, "File %s, line %d: Error(%s): %s\n", filename, linenumber, text, e);
       fclose(f);
 
     }
 
   }
 
-  fprintf(stderr, "Error(%s): %s\n", text, e);
+  fprintf(stderr, "File %s, line %d: Error(%s): %s\n", filename, linenumber, text, e);
 
 }
 
@@ -2079,3 +2081,5 @@ void __afl_coverage_interesting(u8 val, u32 id) {
 
 }
 
+#undef write_error

```

Reply:
Just changed it so.